src/tiles/text-snippet-block.ts

```diff
@@ -0,0 +1,126 @@
+import { join } from 'lit/directives/join.js';
+
+@customElement('text-snippet-block')
+export class TextSnippetBlock extends LitElement {
+  @property({ type: Array }) snippets?: string[] = [];
+
+  @property({ type: String }) viewSize: string = 'desktop';
+
+  render() {
+    const viewSizeClass = this.viewSize === 'grid' ? 'grid' : 'list';
+
+    if (!this.snippets || this.snippets.length === 0) return html`${nothing}`;
+
+    return html`
+      <div id="container" class=${viewSizeClass}>
+        ${this.ellipsisJoinedSnippets}
+      </div>
+
+      ${this.viewSize === 'grid' ? html`<div id="separator"></div>` : nothing}
+    `;
+  }
+
+  /**
+   * An array of HTML templates derived from the snippets, with ellipses inserted
+   * at the beginning, end, and between each pair of snippets.
+   */
+  private get ellipsisJoinedSnippets(): TemplateResult {
+    return html`&hellip; ${join(this.snippetTemplates, html` &hellip; `)}
+    &hellip;`;
+  }
+
+  /**
+   * Returns an array of HTML span templates containing this item's snippets with all of
+   * their `{{{triple-brace-delimited}}}` matches replaced by `<mark>HTML mark tags</mark>`.
+   *
+   * This approach safely avoids the use of `unsafeHTML` and leaves any existing HTML tags
+   * in the snippets intact (as inert text), rather than stripping them away with DOMPurify.
+   *
+   * Note on `<em>` vs. `<mark>`:
+   * The old search page snippets had search keywords demarcated with `<em>` tags.
+   * The `<mark>` tag is semantically more accurate for this use case than `<em>`,
+   * but screen-reader behavior may be different. `<em>` will likely be read in a
+   * different tone, while `<mark>` is often read no differently than ordinary text
+   * in many screen-readers (though there are ways to work around this if needed).
+   */
+  private get snippetTemplates(): TemplateResult[] | undefined {
+    return this.snippets?.map(s => {
+      const matches = s.matchAll(/{{{(.+?)}}}/g);
+      const templates: TemplateResult[] = [];
+
+      // Convert each match into an HTML template that includes:
+      //  - Everything from the end of the previous match (or the beginning of the
+      //      string) up to the current match, as raw text.
+      //  - The current match (excluding the curly braces) wrapped in a `<mark>` tag.
+      let index = 0;
+      for (const match of matches) {
+        if (match.index != null) {
+          templates.push(html`
+            ${s.slice(index, match.index)}
+            <mark>${match[1]}</mark>
+          `);
+          index = match.index + match[0].length;
+        }
+      }
+
+      // Include any text from the last match to the end
+      templates.push(html`${s.slice(index)}`);
+
+      // Squash everything into a single span template
+      return html`<span>${templates}</span>`;
+    });
+  }
+
+  static get styles(): CSSResultGroup {
+    return css`
+      #container {
+        display: -webkit-box;
```

Comment:
Q: do we also need to support other display types here?

Reply:
The `-webkit-line-clamp` approach to ellipsizing the snippet block requires `display: -webkit-box` (and notably, this is widely cross-browser compatible despite the prefix). That said, there are definitely tradeoffs here.

A minor variation from our design spec in this CSS-only approach is that we can't guarantee there will be a space before the end ellipsis. There are some hacky CSS ways around this by abusing float behavior, but those come with their own set of drawbacks. We could also just perform custom truncation via a script and accept the added complexity/performance cost of that.

I suppose if we want the outer container element to have a customizable display type, we could just have a separate inner container that enforces the -webkit-box ellipsis stuff. But my feeling is that we wouldn't really gain much that we can't already do by just setting the display of the host element.